### PR TITLE
Changed oter_t from using bools to usng bitset.

### PR DESCRIPTION
**Pull request by Angular-Angel**
This is just the changes to switch over to the bitset. I'm not entirely sure I did the whole Pull Request propery, so you should probably take a close look at it. It's prety short, so it shouldn't be too much trouble.

### CODE REVIEW

**Comment by Coolthulhu** [comment]
Looks OK to me.
One thing: I don't like that default argument on `set_flag`. I'd make it explicit in all cases.

File: src/omdata.h [hunk 1]
```diff
@@ -22,6 +22,27 @@ struct overmap_spawns {
     int max_population;
     int chance;
 };
+ //terrain flags enum! this is for tracking the indices of each flag.
+    //is_asphalt, is_building, is_subway, is_sewer, is_ants,
```
Comment (BevapDin) [inline src/omdata.h]: Why did you repeat the enum values here?

Comment (Angular-Angel) [inline src/omdata.h]: To make them easier to find for people reading through the code. I'll probabl remove the default argument from the set_flag function, but I'll leave that until the next pull request.

Comment (BevapDin) [inline src/omdata.h]: They are listed *again* right below it as part of the enum definition, why *repeat* them?

If one can read them in the comment, one can also read them in the actual code. Do you really think someone sees them in the comment but is unable to find them in the actual definition? And I usually use the search-in-document function (or similar) of the editor when I need to find some definition.

Comment (KA101) [inline src/omdata.h]: Agreed that in this context the comment-listing is unhelpful.  Something along the lines of "add new terrain flags to this list" (presuming that that is in fact the case) would likely be more helpful.

Comment (Angular-Angel) [inline src/omdata.h]: Oh, that's because I wrote the comment before I wrote the Enum, and never rewrote the comment.

**Comment by Angular-Angel** [comment]
Oh, cool. I had forgotten to go back and update it for compatibility with 0.C, but it looks like you took care of that. Thank you.

**Pull request merged**